src/stratigraphy/benchmark/score.py

```diff
@@ -203,34 +218,64 @@ def create_predictions_objects(
 
 def evaluate(
     predictions,
+    metadata_per_file: BoreholeMetadataList,
     ground_truth_path: Path,
     temp_directory: Path,
     input_directory: Path | None,
     draw_directory: Path | None,
 ):
     """Computes all the metrics, logs them, and creates corresponding MLFlow artifacts (when enabled)."""
-    predictions, number_of_truth_values = create_predictions_objects(predictions, ground_truth_path)
+    #############################
+    # Evaluate the borehole extraction metadata
+    #############################
+    metadata_metrics_list = evaluate_metadata_extraction(metadata_per_file, ground_truth_path)
+    metadata_metrics = metadata_metrics_list.get_cumulated_metrics()
+    document_level_metrics = metadata_metrics_list.get_document_level_metrics()
+
+    document_level_metrics.to_csv(
+        temp_directory / "document_level_metadata_metrics.csv"
+    )  # mlflow.log_artifact expects a file
+
+    # print the metrics
+    logger.info("Metadata Performance metrics:\n\n")
+    logger.info(metadata_metrics)
+
+    if mlflow_tracking:
+        import mlflow
+
+        mlflow.log_metrics(metadata_metrics)
+        mlflow.log_artifact(temp_directory / "document_level_metadata_metrics.csv")
 
-    if input_directory and draw_directory:
-        draw_predictions(predictions, input_directory, draw_directory)
+    #############################
+    # Evaluate the borehole extraction
+    #############################
+    if predictions:
+        predictions, number_of_truth_values = create_predictions_objects(
+            predictions, metadata_per_file, ground_truth_path
+        )
 
-    if number_of_truth_values:  # only evaluate if ground truth is available
-        metrics = evaluate_borehole_extraction(predictions, number_of_truth_values)
+        if input_directory and draw_directory:
+            draw_predictions(predictions, input_directory, draw_directory)
 
-        metrics.document_level_metrics_df().to_csv(
-            temp_directory / "document_level_metrics.csv", index_label="document_name"
-        )  # mlflow.log_artifact expects a file
+        # evaluate the borehole extraction
+        if number_of_truth_values:  # only evaluate if ground truth is available
+            metrics = evaluate_borehole_extraction(predictions, number_of_truth_values)
 
-        metrics_dict = metrics.metrics_dict()
-        logger.info("Performance metrics: %s", metrics_dict)
+            metrics.document_level_metrics_df().to_csv(
+                temp_directory / "document_level_metrics.csv", index_label="document_name"
+            )  # mlflow.log_artifact expects a file
+            metrics_dict = metrics.metrics_dict()
 
-        if mlflow_tracking:
-            import mlflow
+            # Format the metrics dictionary to limit to three digits
+            formatted_metrics = {k: f"{v:.3f}" for k, v in metrics_dict.items()}
+            logger.info("Performance metrics:\n\n %s", formatted_metrics)
```

Comment:
Why the double newline? I thought that log statements should generally be on one line, as many monitoring tools rely on that?

Reply:
Thank you for the explanation, I rolled back on this.